src/StatsdClient/RandomGenerator.cs

```diff
@@ -4,15 +4,26 @@ namespace StatsdClient
 {
     public class RandomGenerator : IRandomGenerator
     {
-        readonly Random _random;
+        [ThreadStatic]
+        static Random _random;
+
+        private static Random Random
+        {
+            get
+            {
+                var random = _random;
+                if (random != null) return random;
+                return _random = new Random(Guid.NewGuid().GetHashCode());

```

Comment:
Well... then you would have the same problem before unless you surrounded it with a lock like so
```c#
lock(_random)
{
    return _random.NextDouble();
}
```